synapse/lib/stormlib/aha.py

```diff
@@ -85,6 +120,50 @@ async def _methAhaGet(self, svcname, filters=None):
         proxy = await self.runt.snap.core.reqAhaProxy()
         return await proxy.getAhaSvc(svcname, filters=filters)
 
+    async def _methCallPeerApi(self, svcname, apiname, timeout=None, skiprun=None):
+        '''
+        Call an API on an AHA service.
+
+        Args:
+            svcname (str): The name of the AHA service to call
+            apiname (str, todo): The API name or todo object
+            timeout (int): Optional timeout in seconds
+            skiprun (str): Optional run ID argument allows skipping self-enumeration.
+        '''
+        svcname = await s_stormtypes.tostr(svcname)
+        apiname = await s_stormtypes.toprim(apiname)
+
+        lib_todo = slib_todo.LibTodo(self.runt)
+        todo = await lib_todo.parse(apiname)
+
+        apiname = (todo.name, todo.args, todo.kwargs)
+
+        timeout = await s_stormtypes.toint(timeout, noneok=True)
+        skiprun = await s_stormtypes.tostr(skiprun, noneok=True)
+
+        proxy = await self.runt.snap.core.reqAhaProxy()
+        svc = await proxy.getAhaSvc(svcname)
+        if svc is None:
+            raise s_exc.NoSuchName(mesg=f'No AHA service found for {svcname}')
+
+        svcinfo = svc.get('svcinfo')
+        svciden = svcinfo.get('iden')
+        if svciden is None:
+            raise s_exc.NoSuchName(mesg=f'Service {svcname} has no iden')
+
+        async for svcname, (ok, info) in proxy.callAhaPeerApi(svciden, apiname, timeout=timeout, skiprun=skiprun):
+            if not ok:
+                if isinstance(info, dict) and 'err' in info:
```

Comment:
This seems like it may duplicate the logic in `synapse.common.result()`. Could we convert to using it?

Reply:
Whoa! I didn't realize that existed, would have saved me some time. :) I'll get it converted.

Reply:
After talking with visi it turns out we don't want to raise these errors directly. I'm going to drop the err extraction code here and update the test cases to validate error propagation from the messages. Will be up in the next commit.